cloudflare/d1/stmt.go

```diff
@@ -59,15 +59,30 @@ func (s *stmt) QueryContext(_ context.Context, args []driver.NamedValue) (driver
 	for i, arg := range args {
 		argValues[i] = arg.Value
 	}
-	resultPromise := s.stmtObj.Call("bind", argValues...).Call("all")
+	resultPromise := s.stmtObj.Call("bind", argValues...).Call("raw", map[string]interface{}{"columnNames": true})
 	rowsObj, err := jsutil.AwaitPromise(resultPromise)
 	if err != nil {
 		return nil, err
 	}
-	if !rowsObj.Get("success").Bool() {
-		return nil, errors.New("d1: failed to query")
+	// If there are no rows to retrieve, length is 0.
+	if rowsObj.Length() == 0 {
+		return &rows{
+			columns: nil,
+			rowsObj: rowsObj,
+		}, nil
+	} else {
```

Comment:
nits: Since the `if` block returns early, there is no need for a corresponding else block. I want to keep the block nesting shallow.

Reply:
thanks.
fixed.